### PR TITLE
Theme example fix and cross browser testing

### PR DESCRIPTION
**Pull request by dbchristopher**
### :sparkles: Changes

- Deep copy every level of theme object for use in slider theme example
- Fix Slider clipping issue in MS/Edge
- Nav appearance fix is MS/Edge

### :white_check_mark: Requirements

- [x] Includes test coverage for all changes
- [x] Build and tests are passing
- [x] PR is ideally < 400LOC

### CODE REVIEW

File: packages/www/src/documentation/components/forms/slider.mdx [hunk 1]
```diff
@@ -86,10 +86,15 @@ You can customize the appearance by modifying the `theme.colors.semanticColors.p
...
+    colors: {
+      ...theme.colors,
+      semanticColors: {
+        ...theme.colors.semanticColors,
+        primary: { ...theme.colors.semanticColors.primary, main: '#008930' },
+      },
+    },
   }
 
-  myTheme.colors.semanticColors.primary.main = '#008930'
```
Comment (dbchristopher) [inline packages/www/src/documentation/components/forms/slider.mdx]: As myTheme was immutably copied on line 88, I actually have no idea how this change was bubbling up to the global scope. But spreading every layer (theme, colors, semantic colors, etc) seems to be necessary in order to contain this color change to the slider example

File: packages/components/src/Sidebar/SidebarGroup.tsx [hunk 2]
```diff
@@ -74,6 +74,8 @@ const SidebarGroupHeading = styled(Heading).attrs({
     display: flex;
     justify-content: space-between;
     width: 100%;
+    background: transparent;
+    border: none;
```
Comment (dbchristopher) [inline packages/components/src/Sidebar/SidebarGroup.tsx]: Fixes issue where some unfortunate default button styling was still showing up in MS browsers:
Screen Shot 2019-12-09 at 1 16 31 PM

File: packages/components/src/Form/Inputs/Slider/Slider.tsx [hunk 3]
```diff
@@ -176,12 +176,13 @@ const sliderThumbCss = css<SliderInputProps>`
 `
 
 const SliderInput = styled.input.attrs({ type: 'range' })<SliderInputProps>`
-  width: 100%;
-  -webkit-appearance: none; /* stylelint-disable-line */
   background: transparent;
   display: block;
+  height: ${({ knobSize }) => knobSize + 6}px;
```
Comment (dbchristopher) [inline packages/components/src/Form/Inputs/Slider/Slider.tsx]: For some reason `overflow: visible` doesn't seem to apply to input styling in MS/Edge. I had to add this input height to allow enough room for the knob+border styling
Screen Shot 2019-12-09 at 1 16 42 PM

File: packages/www/src/documentation/components/forms/slider.mdx (hunk 1 above)
Comment (dbchristopher) [reply packages/www/src/documentation/components/forms/slider.mdx]: Example bugs:
Screen Shot 2019-12-09 at 1 26 53 PM
Screen Shot 2019-12-09 at 1 27 08 PM

Comment (lukelooker) [reply packages/www/src/documentation/components/forms/slider.mdx]: So weird that was happening (can't really understand what's happening but maybe some sort-of cross-talk in React Live. Fix works tho 🤷‍♂

**Pull request merged**